cmd/proxy/main.go

```diff
@@ -202,15 +225,18 @@ func newProxyCommand(p *Proxy, o *ProxyRunOptions) *cobra.Command {
 type Proxy struct {
 }
 
+type StopFunc func()
+
 func (p *Proxy) run(o *ProxyRunOptions) error {
 	o.Print()
 	if err := o.Validate(); err != nil {
 		return fmt.Errorf("failed to validate server options with %v", err)
 	}
+	ctx, cancel := context.WithCancel(context.Background())
 	server := agentserver.NewProxyServer()
 
 	klog.Info("Starting master server for client connections.")
-	err := p.runMasterServer(o, server)
+	masterStop, err := p.runMasterServer(ctx, o, server)
```

Comment:
Is the masterStop a necessary piece for UDS to clean up, or is that a separate improvement?

Reply:
It is necessary for the graceful shutdown of UDS. It also improves the shutdown for grpc but that is a side effect.